pepys_import/core/store/data_store.py

```diff
@@ -1928,43 +1935,53 @@ def split_platform(self, platform_id) -> bool:
             modified=datetime.utcnow(),
             reason=f"Splitting platform: '{platform_id}'.",
         ).change_id
-        datafile_ids = self._find_datafiles_for_platform(platform)
-        for key, value in datafile_ids.items():
-            objects = list(dependent_objects(platform))
+        datafile_ids = self._find_datafiles_and_measurements_for_platform(platform)
+        i = 0
+        percent_per_iteration = 100.0 / len(datafile_ids)
+
+        objects = self._get_comments_and_sensors_of_platform(platform)
+        for key, values in datafile_ids.items():
+            time, measurement_objects = values["time"], values["objects"]
             new_platform = self.add_to_platforms(
                 name=platform.name,
                 nationality=platform.nationality_name,
                 platform_type=platform.platform_type_name,
                 privacy=platform.privacy_name,
                 trigraph=platform.trigraph,
                 quadgraph=platform.quadgraph,
-                identifier=f"{value:%Y%m%d-%H%M%S%f}",
+                identifier=f"{time:%Y%m%d-%H%M%S}-{str(key)[-4:]}",
                 change_id=change_id,
             )
             for obj in objects:
+                primary_key_field = get_primary_key_for_table(obj)
                 if isinstance(obj, Sensor):
-                    sub_objs = list(dependent_objects(obj).limit(1))
-                    if sub_objs and key == sub_objs[0].source_id:
-                        query = self.session.query(Sensor).filter(Sensor.sensor_id == obj.sensor_id)
-                        [
+                    for m_obj in measurement_objects:
+                        if m_obj.sensor_id == obj.sensor_id:
+                            new_sensor = new_platform.get_sensor(
+                                data_store=self,
+                                sensor_name=obj.name,
+                                sensor_type=obj.sensor_type_name,
+                                privacy=obj.privacy_name,
+                                change_id=change_id,
+                            )
+                            old_id = m_obj.sensor_id
+                            m_obj.sensor_id = new_sensor.sensor_id
                             self.add_to_logs(
-                                table=constants.SENSOR,
-                                row_id=s.sensor_id,
-                                field="host",
-                                new_value=str(s.host),
+                                table=obj.__tablename__,
+                                row_id=getattr(obj, primary_key_field),
+                                field="sensor_id",
+                                new_value=str(old_id),
```

Comment:
Why is the new_value being set to the old ID here? Shouldn't it be set to `new_sensor.sensor_id`?

Reply:
I used the same approach we used for merging operations. Used `new_value` field for logging the old values (`sensor_id` or `platform_id`)

Reply:
I recognise the logic here.  There is greater value in storing the old value, since it's the new value that is currently in the database.  If we're in agreement on that we should rename the field (in a different PR).

Reply:
Yes, I think it's more useful when we store the old values.